parser.hpp

```diff
@@ -97,21 +117,32 @@ CONSTEXPR CandyState decode_candy_state(char up_char)
 
 constexpr int candy_size = 3;
 constexpr int candy_type_offset = 1;
+constexpr int row_padding = 5;    // space, pipe, pipe, space, newline
+constexpr int noncandy_rows = 3;  // title, border, border
 
-constexpr auto parse_board_column_count(const constexpr_string_view& str)
+constexpr auto parse_board_parameters(auto&& str)
 {
-    const auto endline = find(str.cbegin(), str.cend(), '\n');
-    return (endline - str.cbegin()) / candy_size;
-}
+    BoardParameters ret{ 0, 0, 0, 0 };
 
-constexpr auto parse_board_row_count(const constexpr_string_view& str)
-{
-    int row_count = count(str.cbegin(), str.cend(), '\n');
+    auto gs_begin = str.cbegin();
+    auto gs_end = str.cend();
+    auto lefttop_corner = find(gs_begin + 1, gs_end, '\n');
+    auto righttop_corner = find(lefttop_corner + 1, gs_end, '\n');
+    auto field_width = righttop_corner - lefttop_corner;
+    ret.column_count = (field_width - row_padding) / candy_size;
+
+    auto score_start = find(gs_begin + 1, gs_end, '>');
+    auto score_start_position = score_start - gs_begin;
+    ret.row_count = ((score_start_position / field_width) - noncandy_rows) / candy_size;
 
-    if (str[str.size() - 1] != '\n')
-        ++row_count;
+    const auto score_value_offset = 9;  // length of "> score: "
+    ret.score = extract_int(score_start + score_value_offset);
 
-    return row_count / candy_size;
+    auto moves_start = find(score_start + 1, gs_end, '>');
+    const auto moves_value_offset = 9;  // length of "> moves: "
+    ret.moves = extract_int(moves_start + moves_value_offset);
+
+    return ret;
 }
 
 constexpr auto isolate_board_game(auto&& str)
```

Comment:
I am guessing that this function has no use anymore. We could remove it.

Reply:
Removed :-)